providers/default/defltprov.c

```diff
@@ -51,7 +51,42 @@ static int deflt_get_params(const OSSL_PROVIDER *prov,
 }
 
 static const OSSL_ALGORITHM deflt_digests[] = {
+    { "SHA1", "default=yes", sha1_functions },
```

Comment:
Is the FIPS work a chance to make the various SHA algorithms optional?

Reply:
There are still cases where SHA1 needs to be used though right?

Reply:
e.g FIPS-186-4 still uses SHA1 to support verification of old signatures.

Reply:
I wasn't thinking FIPS.  Currently there are several algorithms it is impossible to build lib crypto without.  SHA1 is one of them.  I also note that SHA3 is another that was relatively recently added :(